analysis/hooks.py

```diff
@@ -412,13 +416,23 @@ def run(self, Object, HandleAttributes, PassedAccessState, DesiredAccess, Object
             # HandleAttributes is not OBJ_FORCE_ACCESS_CHECK.
             tmp_state = self.state.copy()
             tmp_state.solver.add(HandleAttributes & 1024 == 0)
-
+            handle = claripy.BVS(f"ObReferenceObjectByHandle_{hex(self.state.callstack.ret_addr)}", self.state.arch.bits)
+            self.state.memory.store(Handle, handle,  self.state.arch.bytes, endness=self.state.arch.memory_endness, disable_actions=True, inspect=False)
             # Check if we can control the parameters of ObOpenObjectByPointer.
             if tmp_state.satisfiable() and str(Object) in self.state.globals['tainted_eprocess']:
+                self.state.globals['tainted_handles'] = (str(handle), )
                 ret_addr = hex(self.state.callstack.ret_addr)
                 utils.print_vuln('controllable process handle', 'ObOpenObjectByPointer - Object controllable', self.state, {'Object': str(Object), 'Handle': str(Handle)}, {'return address': ret_addr})
         return 0
 
+class HookZwTerminateProcess(angr.SimProcedure):
+    def run(self, ProcessHandle, ExitStatus):
+        ret_addr = hex(self.state.callstack.ret_addr)
+        if str(ProcessHandle) in self.state.globals['tainted_handles']:
+            utils.print_vuln('controllable ZwTerminateProcess', 'ZwTerminateProcess - handle controllable', self.state, {'ProcessHandle': str(ProcessHandle)}, {'return address': ret_addr})
```

Comment:
The title of this vulnerability should be what you write on README: `arbitrary process termination`